### DISCUSSION TITLE
Localization: Ignore browser language (always use default culture)

### DISCUSSION THREAD

**Discussion by XpandIT-RFID** [OrchardCMS/OrchardCore#10149] [Q&A]
Hi, I am currently developing a website with the Orchard Core CMS. The website is supposed to only be available in the pt-PT language.
I am trying to create a pt-PT.po to override some english texts, for example the errors in the Login Controller.
The issue is that even if I set the localization to only use pt-PT it ignores it because the browser default culture is English..

Is there a way to force this?
cc: @Skrypt , @deanmarcussen

![image](https://user-images.githubusercontent.com/83280308/130229733-8cd457b1-0774-424a-b0eb-6d03fe04ea3e.png)
![image](https://user-images.githubusercontent.com/83280308/130229762-5b12999a-f6a7-4b1d-ac0a-3d01c47c0213.png)

**Comment by hishamco** [comment]
I frequently saw similar issue, I'm thinking it's a cookies issue /cc @jptissot

**Comment by XpandIT-RFID** [comment]
Solved it with this code on the .web project Startup:

pt-PT seems to not work.

```
services.Configure<RequestLocalizationOptions>(options =>
{
    var defaultLang = "pt-BR";
    var supportedCultures = new List<CultureInfo>
    {
        new CultureInfo(defaultLang),
        new CultureInfo(defaultLang)
    };

    options.DefaultRequestCulture = new RequestCulture(culture: defaultLang, uiCulture: defaultLang);
    options.SupportedCultures = supportedCultures;
    options.SupportedUICultures = supportedCultures;

    options.RequestCultureProviders.Insert(0, new CustomRequestCultureProvider(context =>
    {
        return Task.FromResult(new ProviderCultureResult(defaultLang, defaultLang));
    }));
});
```

**Reply by hishamco** [reply]
No need to set the default culture from here, while the `LocalizationService` will take care about what you did above

**Reply by Skrypt** [reply]
I don't see any issue there. It works as intended. Your browser is set to "en" before pt-PT so it should display in english. The only thing though is that you might want to **unregister then the AcceptLanguageHeaderRequestCultureProvider simply if it is of no use for you**. No need to do what you did there in your custom code.

Here, you need to know that if your OS system culture is set as "en" ; that culture will be added in the supported cultures. So, this is probably why it is allowing to be displayed in "en" in your case. And the system culture should always be a supported culture. 

See : https://github.com/OrchardCMS/OrchardCore/blob/ac7598ceaf54f5c5a5fed01e8fd0a2c3a7f2f3a3/src/OrchardCore/OrchardCore/Localization/DefaultLocalizationService.cs#L12

So, your question is, how do I always return my website in "pt-PT" culture? I believe that the only thing that can change the culture of the website without being necessarly wanted is the AcceptLanguageHeaderRequestCultureProvider. This has nothing to do with the supported cultures here because your system culture should be supported.
